docs/TELEMETRY.md

```diff
@@ -0,0 +1,131 @@
+
+-  Dimensions are environment information, including browser capabilities and options passed to Web Chat
+-  Events will include a name, optional string data, and optional non-negative finite integer value
+-  Exceptions will include the JavaScript error object
+-  Timings will fire two events, `timingstart` and `timingend`
+   -  `timingstart` will include a name
+   -  `timingend` will include a name, and duration measured in milliseconds
+
+Feature updates on minor versions may introduce new or remove outdated measurements. To better understand user behaviors with different set of measurements, developers are advised to tag their telemetry data with Web Chat version.
+
+### Dimensions
+
+The following information will be emitted on every measurement. Dimensions may change during the session.
+
+| Name                                   | Description                                                                              |
+| -------------------------------------- | ---------------------------------------------------------------------------------------- |
+| `prop:locale`                          | Locale specified                                                                         |
+| `prop:locale:numChange`                | Number of times locale has changed                                                       |
+| `prop:speechRecognition`               | `"false"` if speech recognition is switched off                                          |
+| `prop:speechSynthesis`                 | `"false"` if speech synthesis is switched off                                            |
+| `capability:downscaleImage:workerType` | `"web worker"` if the browser support Web Worker and offline canvas, otherwise, `"main"` |
+
+> `prop:speechRecognition` and `prop:speechSynthesis` does not represents browser capabilities on Speech Recognition and Speech Synthesis.
+
+Some telemetry services may have limited number of dimensions. For example, Google Analytics has a limit of 20 custom dimensions. As Web Chat may introduce more dimensions later, developers are advised to pick dimensions they needed before sending to their services.
+
+### Events for hooks
+
+When the following hooks are called, one or more event measurements will be emitted.
+
+| Hooks              | Events                  | Description                                               |
+| ------------------ | ----------------------- | --------------------------------------------------------- |
+| `useSendFiles`     | `sendFiles`             | Emit when the user uploading files                        |
+| `useSendFiles`     | `sendFiles:numFile`     | Number of files uploaded                                  |
+| `useSendFiles`     | `sendFiles:sumSizeInKB` | Total file size in kilobytes                              |
+| `useSubmitSendBox` | `submitSendBox`         | `data` is the method of submit, for example, `"keyboard"` |
```

Comment:
should all examples be listed. For example mouse?

Reply:
Currently, we only added a few measurements, no mouse yet.

When we talk to legal team later, we will know how many measurements we could set.